sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/implementation/signature/AbstractKeyVaultKeyLessSignature.java

```diff
@@ -0,0 +1,116 @@
+
+    public AbstractKeyVaultKeyLessSignature() {
+        this.keyVaultClient = createKeyVaultClientBySystemProperty();
+    }
+
+    void setKeyVaultClient(KeyVaultClient keyVaultClient) {
+        this.keyVaultClient = keyVaultClient;
+    }
+
+    // After throw UnsupportedOperationException, other methods will be called.
+    // such as RSAPSSSignature#engineInitVerify.
+    @Override
+    protected void engineInitVerify(PublicKey publicKey) {
+        throw new UnsupportedOperationException("engineInitVerify() not supported");
+    }
+
+    // After throw UnsupportedOperationException, other methods will be called.
+    // such as RSAPSSSignature#engineVerify.
+    @Override
+    protected boolean engineVerify(byte[] signature) {
+        throw new UnsupportedOperationException("getParameter() not supported");
+    }
+
+    // After throw UnsupportedOperationException, other methods will be called.
+    // such as ECDSASignature#engineGetParameter.
+    @Override
+    @Deprecated
+    protected Object engineGetParameter(String param)
+        throws InvalidParameterException {
+        throw new UnsupportedOperationException("getParameter() not supported");
+    }
+
+    // After throw UnsupportedOperationException, other methods will be called.
+    // such as RSAPSSSignature#engineSetParameter.
+    @Override
+    @Deprecated
+    protected void engineSetParameter(String param, Object value)
+        throws InvalidParameterException {
+        throw new UnsupportedOperationException("setParameter() not supported");
+    }
+
+    // After throw UnsupportedOperationException, other methods will be called.
+    // such as RSAPSSSignature#engineInitSign.
+    @Override
+    protected void engineInitSign(PrivateKey privateKey, SecureRandom random) {
+        if (privateKey instanceof KeyVaultPrivateKey) {
+            keyId = ((KeyVaultPrivateKey) privateKey).getKid();
+        } else {
+            throw new UnsupportedOperationException("engineInitSign() not supported which private key is not instance of KeyVaultPrivateKey");
+        }
+    }
+
+    @Override
+    protected void engineInitSign(PrivateKey privateKey) {
+        engineInitSign(privateKey, null);
+    }
+
+    /**
+     * Get the message digest value.
+     * @return the message digest value.
+     */
+    protected byte[] getDigestValue() {
+        return messageDigest.digest();
+    }
+
+    @Override
+    protected void engineUpdate(byte b) {
+        this.messageDigest.update(b);
+    }
+
+    @Override
+    protected void engineUpdate(byte[] b, int off, int len) {
+        this.messageDigest.update(b, off, len);
+    }
+
+    @Override
+    protected void engineUpdate(ByteBuffer b) {
+        this.messageDigest.update(b);
+    }
+
+    //Override this method just do not throw an exception to enable this signature
+    @Override
+    protected void engineSetParameter(AlgorithmParameterSpec params) throws InvalidAlgorithmParameterException {
```

Comment:
How about adding `throw new UnsupportedOperationException("setParameter() not supported");` to avoid forgetting override this method.

Reply:
According to my test results, it is sufficient to execute the empty method in `KeyVaultKeyLessECSignature`. Do I need to add throw here and add an empty method in the subclass respectively

Reply:
OK. then keep current code.